yapapi/executor/__init__.py

```diff
@@ -185,14 +185,36 @@ async def run_worker(work_context: WorkContext) -> None:
             with work_queue.new_consumer() as consumer:
                 try:
 
-                    async def task_generator() -> AsyncIterator[Task[D, R]]:
+                    # the `task_generator` here is what feeds the user's `worker` directly
+                    #
+                    # by turning it into an actual `AsyncGenerator`, we're able
```

Comment:
What do you mean by "by turning it into"?
I understand this from the POV of this PR, but it will be confusing in the future.

IOW, a comment should describe "why we have this" and maybe sometimes "what we have", but should not reference any previous states.

Reply:
okay, got the point, will fix it :)